### ISSUE TITLE
Is there any plan to make components reactive?

### ISSUE THREAD

**Issue by csyszf** [NG-ZORRO/ng-zorro-antd#136]
## I'm submitting a...

<pre><code>
[ ] Bug report  
[x] Feature request
[ ] Documentation issue or request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Support request => Please do not submit support request here
</code></pre>

## Current behavior

These components current are barely reactive, so some components won't work well if used by parent/root components which marked `ChangeDetectionStrategy.OnPush`, unless mark changes manually.

## Expected behavior

Components work well even if they're under `ChangeDetectionStrategy.OnPush` scene.

## Minimal reproduction of the problem with instructions

1. Create a component with ChangeDetectionStrategy.OnPush
```typescript
@Component({
  selector: 'app-root',
  changeDetection: ChangeDetectionStrategy.OnPush,
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'app';
}
```
2. Add `<nz-popconfirm>` to the component
```html
  <nz-popconfirm [nzTitle]="'确定要删除吗？'" (nzOnConfirm)="confirm()"
    (nzOnCancel)="cancel()">
    <a nz-popconfirm>删除</a>
  </nz-popconfirm>
```
3. Click the a element and there's no popup

## Environment

<pre><code>
Angular version: X.Y.Z

ng-zorro-antd version: X.Y.Z

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] IE version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

----------------

I've noticed that comment https://github.com/NG-ZORRO/ng-zorro-antd/issues/74#issuecomment-323240531

For everyone who uses OnPush mode, they must add `ref.markForCheck()` to every logic which should update components' data, or turn to use Default mode.

Use rxjs/Observable could solve this problem, But i don't know whether it will lead to more complex problems, I've not tested for ng-zorro yet.

But i think it would not break the current component structure - it is only need to change the middle layer data to observable, like the `nzVisible` property in [`NzPopconfirmComponent`](https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/src/components/popconfirm/nz-popconfirm.component.ts), the other Input properties will not be changes.

Anyway thanks for your nice work, i really like those components and theirs style.

**Comment by vthinkxie** [comment]
Yes, onPush is not supported yet, and I can't make a promise when it will be supported at this moment, because the most important thing is ensuring all bugs to be fixed when 0.5.0 final version release.
It needs a lot of work to support onPush since that all components need to be tested, and some strategies need to be rewrite.

welcome pr for this, thanks for your feedback.

**Comment by LinBoLen** [comment]
+1 for manual detectChanges, markForCheck
